src/Sylius/Bundle/InstallerBundle/Command/SetupCommand.php

```diff
@@ -115,28 +124,46 @@ protected function setupAdministratorUser(InputInterface $input, OutputInterface
      * @param InputInterface  $input
      * @param OutputInterface $output
      */
-    protected function setupCurrency(InputInterface $input, OutputInterface $output)
+    protected function setupLocale(InputInterface $input, OutputInterface $output)
     {
-        $currencyRepository = $this->get('sylius.repository.currency');
-        $currencyManager = $this->get('sylius.manager.currency');
-        $currencyFactory = $this->get('sylius.factory.currency');
+        $localeRepository = $this->get('sylius.repository.locale');
+        $localeManager = $this->get('sylius.manager.locale');
+        $localeFactory = $this->get('sylius.factory.locale');
 
-        do {
-            $code = $this->getCurrencyCode($input, $output);
+        $code = trim($this->getContainer()->getParameter('sylius.locale'));
+        $name = Intl::getLanguageBundle()->getLanguageName($code);
+        $output->writeln(sprintf('Adding <info>%s</info> locale.', $name));
 
-            $valid = true;
+        if (null !== $existingLocale = $localeRepository->findOneBy(['code' => $code])) {
+            $this->locale = $existingLocale;
 
-            if (0 !== count($errors = $this->validate(trim($code), [new Currency()]))) {
-                $valid = false;
-            }
+            $localeManager->flush();
+
+            return;
+        }
 
+        $locale = $localeFactory->createNew();
+        $locale->setCode($code);
 
-                $this->writeErrors($output, $errors);
-        } while (!$valid);
+        $localeManager->persist($locale);
+        $localeManager->flush();
+
+        $this->locale = $locale;
+    }
+
+    /**
+     * @param InputInterface  $input
+     * @param OutputInterface $output
+     */
+    protected function setupCurrency(InputInterface $input, OutputInterface $output)
+    {
+        $currencyRepository = $this->get('sylius.repository.currency');
+        $currencyManager = $this->get('sylius.manager.currency');
+        $currencyFactory = $this->get('sylius.factory.currency');
 
-        $code = trim($code);
+        $code = trim($this->getContainer()->getParameter('sylius.currency'));
         $name = Intl::getCurrencyBundle()->getCurrencyName($code);
-        $output->writeln(sprintf('Adding <info>%s</info>', $name));
+        $output->writeln(sprintf('Adding <info>%s</info> currency.', $name));
 
         if (null !== $existingCurrency = $currencyRepository->findOneBy(['code' => $code])) {
```

Comment:
Before creating/setting a base currency, we should find current base currency and if exist `setBase(false)` to avoid having many base currencies in the database.